src/pages/AMM/Swap/components/SwapForm/SwapForm.tsx

```diff
@@ -0,0 +1,183 @@
+import { CurrencyExt } from '@interlay/interbtc-api';
+import { ChangeEventHandler, FormEventHandler, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { toast } from 'react-toastify';
+import { useDebounce } from 'react-use';
+
+import { StoreType } from '@/common/types/util.types';
+import { Card, CardProps, Divider, Flex, H1, TokenInput } from '@/component-library';
+import { AuthCTA } from '@/components/AuthCTA';
+import { SwapPair, SwapSlippage } from '@/types/swap';
+import { useGetCurrencies } from '@/utils/hooks/api/use-get-currencies';
+
+import { useSwapFormData } from '../../hooks/use-swap-form-data';
+import { SlippageManager } from '../SlippageManager';
+import { SwapInfo } from '../SwapInfo';
+import { SwapDivider } from './SwapDivider';
+
+// TODO: here we will call the lib to get the expected output
+const getOutput = async (param?: number) => {
+  return param;
+};
+
+const getPairChange = (pair: SwapPair, currency: CurrencyExt, name: string): SwapPair => {
+  switch (name) {
+    case FormFields.INPUT_TICKER: {
+      if (currency.ticker === pair.output?.ticker) {
+        return { input: currency, output: pair.input };
+      }
+
+      return { ...pair, input: currency };
+    }
+    case FormFields.OUTPUT_TICKER: {
+      if (currency.ticker === pair.input?.ticker) {
+        return { input: pair.output, output: currency };
+      }
+
+      return { ...pair, output: currency };
+    }
+    default:
+      return pair;
+  }
+};
+
+enum FormFields {
+  INPUT_AMOUNT = 'input-amount',
+  INPUT_TICKER = 'input-ticker',
+  OUTPUT_AMOUNT = 'output-amount',
+  OUTPUT_TICKER = 'output-ticker'
+}
+
+type SwapFormData = {
+  [FormFields.INPUT_AMOUNT]: { value: string };
+  [FormFields.INPUT_TICKER]: { value: string };
+  [FormFields.OUTPUT_AMOUNT]: { value: string };
+  [FormFields.OUTPUT_TICKER]: { value: string };
+};
+
+type Props = {
+  pair: SwapPair;
+  onChangePair: (pair: SwapPair) => void;
+};
+
+type InheritAttrs = CardProps & Props;
+
+type SwapFormProps = Props & InheritAttrs;
+
+const SwapForm = ({ pair, onChangePair, ...props }: SwapFormProps): JSX.Element | null => {
+  const { t } = useTranslation();
+
+  const { bridgeLoaded } = useSelector((state: StoreType) => state.general);
+  const { getCurrencyFromTicker } = useGetCurrencies(bridgeLoaded);
+
+  const [slippage, setSlippage] = useState<SwapSlippage>('0.1%');
```

Comment:
Would be a bit tidier to have the instances of useState grouped together at the top of component, above the hooks (i.e. moved to L69).